src/codec.ts

```diff
@@ -16,7 +14,7 @@ const version = 2;
 /**
  * Used internally in the pairing object to represent a handshake message
  */
-export class NoiseHandshakeMessage extends MessageV0 implements Message {
+export class NoiseHandshakeMessage extends DecodedMessage {
```

Comment:
`implements IDecodedMessage`, see my comment on the encoder.

Reply:
I will follow same approach with `extends ... implements` here.